src/main/java/de/retest/web/RecheckSeleniumAdapter.java

```diff
@@ -61,18 +63,52 @@ public boolean canCheck( final Object toVerify ) {
 		logger.info( "Retrieving attributes for each element." );
 		final Set<String> cssAttributes = attributesProvider.getCssAttributes();
 		final JavascriptExecutor jsExecutor = (JavascriptExecutor) driver;
-		@SuppressWarnings( "unchecked" ) final Map<String, Map<String, Object>> result =
+		@SuppressWarnings( "unchecked" )
+		final Map<String, Map<String, Object>> rawAttributesMapping =
 				(Map<String, Map<String, Object>>) jsExecutor.executeScript( getQueryJS(), cssAttributes );
 
-		logger.info( "Checking website {} with {} elements.", driver.getCurrentUrl(), result.size() );
-		final RootElement lastChecked = convertToPeers( result, driver.getTitle(), shoot( driver ) );
+		logger.info( "Checking website {} with {} elements.", driver.getCurrentUrl(), rawAttributesMapping.size() );
+		final RootElement lastChecked = convertToPeers( rawAttributesMapping, driver.getTitle(), shoot( driver ) );
+
+		addChildrenFromFrames( driver, cssAttributes, lastChecked );
+
 		if ( driver instanceof UnbreakableDriver ) {
 			((UnbreakableDriver) driver).setLastActualState( lastChecked );
 		}
 
 		return Collections.singleton( lastChecked );
 	}
 
+	public void addChildrenFromFrames( final WebDriver driver, final Set<String> cssAttributes,
+			final RootElement lastChecked ) {
+		final JavascriptExecutor jsExecutor = (JavascriptExecutor) driver;
+		final List<Element> frames = de.retest.web.selenium.By.findElements( lastChecked.getContainedElements(),
+				element -> "iframe".equalsIgnoreCase( element.getIdentifyingAttributes().getType() )
```

Comment:
If we 

* extract the predicate
* and use an import for `By`

I think it would be a bit more readable:

```java
final Predicate<Element> isFrame = element -> {
    String type = element.getIdentifyingAttributes().getType();
    return Stream.of( "iframe", "frame" ).anyMatch( type::equalsIgnoreCase );
}
final List<Element> frames = By.findElements( lastChecked.getContainedElements(), isFrame );
```